state/integration_test.go

```diff
@@ -0,0 +1,135 @@
+package state
+
+import (
+	"context"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/suite"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	"google.golang.org/grpc"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/testutil/testnode"
+	blobtypes "github.com/celestiaorg/celestia-app/x/payment/types"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+func TestIntegrationTestSuite(t *testing.T) {
+	suite.Run(t, new(IntegrationTestSuite))
+}
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	cleanups []func() error
+	accounts []string
+	cctx     testnode.Context
+
+	accessor *CoreAccessor
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	if testing.Short() {
+		s.T().Skip("skipping test in unit-tests or race-detector mode.")
+	}
+
+	s.T().Log("setting up integration test suite")
+	require := s.Require()
+
+	// we create an arbitrary number of funded accounts
+	for i := 0; i < 300; i++ {
```

Comment:
nit: we create only ten in Swamp because creating those accounts tends to be very slow. You test it out yourself, but I don't wanna block release with it.

Reply:
yeah, this was just a remnant of debugging, where we didn't know which addresses where failing. Sometimes we would get 10 addresses that work fine, but never 300. I should have used specifc priv keys that have addresses that fail